Mathlib/AlgebraicTopology/SimplicialSet/StrictSegal.lean

```diff
@@ -0,0 +1,123 @@
+/-
+Copyright (c) 2024 Emily Riehl. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Mario Carneiro, Emily Riehl, Joël Riou
+-/
+import Mathlib.AlgebraicTopology.SimplicialSet.Nerve
+import Mathlib.AlgebraicTopology.SimplicialSet.Path
+import Mathlib.CategoryTheory.Functor.KanExtension.Adjunction
+import Mathlib.CategoryTheory.Functor.KanExtension.Basic
+
+/-!
+# Strict Segal simplicial sets
+
+A simplicial set `X` satisfies the `StrictSegal` condition if for all `n`, the map
+`X.spine n : X _[n] → X.Path n` is a bijection.
+
+Examples of `StrictSegal` simplicial sets are given by nerves of categories.
+
+TODO: Show that these are the only examples: that a `StrictSegal` simplicial set is isomorphic to
+the nerve of its homotopy category.
+
+`StrictSegal` simplicial sets have an important property of being 2-coskeletal which is proven
+in `Mathlib.AlgebraicTopology.SimplicialSet.Coskeletal`.
+-/
+
+universe v u
+
+open CategoryTheory
+
+open Simplicial SimplexCategory
+
+namespace SSet
+
+variable (X : SSet.{u})
+
+/-- A simplicial set `X` satisfies the strict Segal condition if its simplices are uniquely
+determined by their spine. -/
+class StrictSegal : Prop where
+  segal : ∀ (n : ℕ), Function.Bijective (X.spine n)
+
+variable {X} in
+/-- The diagonal of a simplex is the long edge of the simplex.-/
+def diagonal {n : ℕ} (Δ : X _[n]) : X _[1] := X.map ((mkOfDiag n).op) Δ
+
+namespace StrictSegal
+variable {X : SSet.{u}} [StrictSegal X] {n : ℕ}
+
+/-- In the presence of the strict Segal condition, a path of length `n` extends to an `n`-simplex
+whose spine is that path. -/
+noncomputable def spineToSimplex : Path X n → X _[n] :=
+  (Equiv.ofBijective _ (segal n)).invFun
+
+@[simp]
+theorem spineToSimplex_spine (f : Path X n) :
+    X.spine n (StrictSegal.spineToSimplex f) = f :=
+  (Equiv.ofBijective _ (segal n)).right_inv f
+
+@[simp]
+theorem spineToSimplex_vertex (i : Fin (n + 1)) (f : Path X n) :
+    X.map (const [0] [n] i).op (spineToSimplex f) = f.vertex i := by
+  rw [← spine_vertex, spineToSimplex_spine]
+
+@[simp]
+theorem spineToSimplex_spine_edge (i : Fin n) (f : Path X n) :
```

Comment:
The following name would be a little bit more coherent:
```suggestion
theorem spineToSimplex_arrow (i : Fin n) (f : Path X n) :
```

Reply:
done